### ISSUE TITLE
No Icons in Homeassistant App on Android

### ISSUE THREAD

**Issue by podkilla** [arallsopp/hass-hue-icons#58]
Thanks for your work - icons work fine in my browser.

Are those icons supposed to work in the Homeassistant Android App as well?

**Comment by arallsopp** [comment]
They should do. They work on the iOS app (only one I have to test).

![image](https://user-images.githubusercontent.com/2462104/132916771-c9a55e6b-0360-468d-90c7-27d964422db7.png)

Few questions to see if we can pinpoint where it might be going wrong:

- Did you install via HACS?
- Do you see the icons if you load an alternate dashboard page, then come back?
- Did you add the extra_module_url parameter in your configuration.yaml
- Are you able to try 'App Configuration -> Debugging -> Reset frontend cache'
- Have you force closed the app and restarted it?
- Do they work on a desktop?

**Comment by podkilla** [comment]
> Few questions to see if we can pinpoint where it might be going wrong:
> 
>     * Did you install via HACS?
> 
Yes, i did.

>     * Do you see the icons if you load an alternate dashboard page, then come back?
> 
Not on my phone but in my browser yes.

>     * Did you add the extra_module_url parameter in your configuration.yaml
> 
Yes, i added this to my configuration files 
```
frontend:
   extra_module_url:
     - /hacsfiles/hass-hue-icons/hass-hue-icons.js
```
 
>     * Are you able to try 'App Configuration -> Debugging -> Reset frontend cache'
> 
I dont have that option in Homeassistant App v. 2021.9.0-minimal

>     * Have you force closed the app and restarted it?
> 
Yes, and i emptied the cache as well.

>     * Do they work on a desktop?
They work fine on my desktop.

**Comment by arallsopp** [comment]
Perfect response, thanks for that.

OK. It sounds like you've got it installed correctly, and that your issue is specific to the android app. I don't have the android app, so can't test it, but wouldn't expect it to behave differently from iOS. I'll have a nose around and see if I can find more information for you.

**Comment by arallsopp** [comment]
just to rule out anything daft, does the android stock browser / chrome see the icons?

**Comment by podkilla** [comment]
I just checked and they are shown in my vanadium browser v93.0.4577.62 on my android device.

**Comment by arallsopp** [comment]
Ok. That's good to know. So it's just to do with the app then, for sure. 

https://www.reddit.com/r/homeassistant/comments/nkwdrv/icons_not_loading_on_mobile_app_or_browser_unless/?utm_source=share&utm_medium=ios_app&utm_name=iossmf suggests "View -> Reload Page" sorts their missing icons on the app. Does that work for you? Is that an option in the app?

**Comment by podkilla** [comment]
Yes, i have the three dotted menu and already tried to reload page and resources. Works on desktop all the time, not in the app though.

**Comment by arallsopp** [comment]
Sounds like you're doing all the right things. I'm sorry it's not working for you. I don't know enough about the android app to understand why it might be different from your browser. I'll keep looking but it's just me and google right now :)

**Comment by arallsopp** [comment]
Does this help? 

https://community.home-assistant.io/t/icons-doesnt-display-on-app/251852

**Comment by podkilla** [comment]
No, sadly it does not. I dont have the option to clear the cache within the app. Clearing the cache from android has no effect. 
I just tested another android phone with lineageOS installed Homeassistant App  v.2021.9.0-full. Still no option to clear the cache from within the app and the icons still not showing.

**Comment by arallsopp** [comment]
Hmm. That's a pain. Sorry. I'll see if I can borrow an android phone and work out if there's a way to see what it's up to.

**Comment by podkilla** [comment]
I had an update of a repo in HACS today. I updated from one of my android phones. After the update the usual HACs dialogue pops up that reminds to reload the UI. I pressed reload and suddenly the icons are visible on this device only though. On the other device the icons are still missing.
This reload seems to be something different than the one behind the three dotted menu on the dashboard.

So i guess this issue can be closed now. Thanks for the help.
